src/bp/ui-admin/src/Auth/index.js

```diff
@@ -25,44 +34,63 @@ export function logout() {
 }
 
 export default class BasicAuthentication {
-  login = async ({ email, password, newPassword }) => {
+  login = async ({ email, password, newPassword }, loginUrl) => {
     if (this.isAuthenticated()) {
       return
     }
-    await this.doLogin({ email, password, newPassword })
-  }
 
-  async doLogin({ email, password, newPassword }) {
-    const { data } = await api.getAnonymous({ toastErrors: false }).post('/auth/login', {
+    const { data } = await api.getAnonymous({ toastErrors: false }).post('/auth' + loginUrl, {
       email,
       password,
       newPassword
     })
 
-    this.setSession({ expiresIn: 7200, idToken: data.payload.token })
+    const { token } = data.payload
+    this.setSession({ expiresIn: 7200, idToken: token })
+
+    await this.setupWorkspace()
 
     const returnTo = history.location.query.returnTo
-    if (returnTo) {
-      window.location.replace(returnTo)
-    } else {
-      history.replace(HOME_ROUTE)
+    returnTo ? window.location.replace(returnTo) : history.replace(HOME_ROUTE)
+  }
+
+  getStrategyConfig = async userStrategy => {
+    const { data } = await api.getAnonymous().get('/auth/config')
+    if (!data.payload || !data.payload.strategies || !data.payload.strategies.length) {
+      return
     }
+
+    const { strategies, isFirstUser } = data.payload
+
+    const strategyId = userStrategy || strategies[0].strategyId
+    const strategyConfig = strategies.find(s => s.strategyId === strategyId)
+
+    return strategyConfig && { ...strategyConfig, isFirstUser }
   }
 
-  register = async ({ email, password }) => {
+  setupWorkspace = async () => {
+    const { data: workspaces } = await api.getSecured().get('/auth/me/workspaces')
+    if (!workspaces || !workspaces.length) {
+      throw new Error(`You must have access to at least one workspace to login.`)
+    }
+
+    // We set either the active workspace, or the first in the list he's allowed otherwise.
+    setActiveWorkspace(getActiveWorkspace() || workspaces[0].workspace)
```

Comment:
so we don't display available workspaces for the user at first login  (or when ls is cleared) ?

Reply:
Nop, not for now